src/dashboards/components/RefreshContext.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {createContext, useReducer, useCallback, FC} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
+import moment from 'moment'
+
+// Types
+import {
+  CustomTimeRange,
+  AutoRefreshStatus,
+  AutoRefresh,
+  AppState,
+} from 'src/types'
+
+// Actions
+import {
+  setAutoRefreshInterval,
+  setAutoRefreshStatus,
+  setAutoRefreshDuration,
+  setInactivityTimeout,
+} from 'src/shared/actions/autoRefresh'
+
+export const AutoRefreshContext = createContext(null)
+
+interface AutoRefreshState {
+  duration: CustomTimeRange
+  inactivityTimeout: string
+  inactivityTimeoutCategory: string
+  refreshMilliseconds: AutoRefresh
+}
+
+const calculateTimeout = (timeout, timeoutUnit) => {
+  const timeoutNumber = parseInt(timeout)
+  const startTime = moment(new Date())
+  const copyStart = startTime.unix()
```

Comment:
```suggestion
  const startTime = moment().unix()
  const copyStart = startTime
```

Reply:
I need this to stay the way it is bc of the next line